File: scrape_assets.rb
require 'open-uri'
require 'fileutils'
require 'uri'

# Configuration
BASE_URL = "https://www.anantawealth.com"
INPUT_FILE = "original_index.html"
OUTPUT_FILE = "index.html"
ASSET_DIRS = {
  "css" => "css",
  "js" => "js",
  "image" => "images",
  "font" => "fonts"
}

# Create directories
ASSET_DIRS.values.each do |dir|
  FileUtils.mkdir_p(dir)
end

def download_file(url, folder, default_ext="")
  begin
    # Handle protocol-relative URLs
    if url.start_with?("//")
      url = "https:" + url
    end

    # Absolute URL
    full_url = URI.join(BASE_URL, url).to_s
    
    # Parse filename
    parsed = URI.parse(full_url)
    filename = File.basename(parsed.path)
    
    # Handle missing filename
    if filename.nil? || filename.empty? || filename == "/"
      return url
    end

    # Clean filename
    filename = filename.gsub(/[<>:"\/\\|?*]/, '_')
    
    if File.extname(filename).empty? && !default_ext.empty?
      filename += default_ext
    end

    local_path = File.join(folder, filename)
    
    # Check if exists
    if File.exist?(local_path)
      # puts "Skipping existing: #{filename}"
      return "#{folder}/#{filename}"
    end

    puts "Downloading: #{full_url} -> #{local_path}"
    
    # Download
    URI.open(full_url, "User-Agent" => "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7)") do |image|
      File.open(local_path, "wb") do |file|
        file.write(image.read)
      end
    end
    
    return "#{folder}/#{filename}"
  rescue => e
    puts "Failed to download #{url}: #{e.message}"
    return url
  end
end

content = File.read(INPUT_FILE)

# 1. CSS Links
content = content.gsub(/<link[^>]+href=["']([^"']+)["'][^>]*>/) do |match|
  full_tag = match
  href = $1
  
  if full_tag.include?('rel="stylesheet"') || full_tag.include?("rel='stylesheet'") || href.end_with?('.css')
     new_path = download_file(href, ASSET_DIRS["css"], ".css")
     full_tag.sub(href, new_path)
  else
    full_tag
  end
end

# 2. Scripts
content = content.gsub(/<script[^>]+src=["']([^"']+)["'][^>]*>/) do |match|
  full_tag = match
  src = $1
  new_path = download_file(src, ASSET_DIRS["js"], ".js")
  full_tag.sub(src, new_path)
end

# 3. Images
content = content.gsub(/<img[^>]+src=["']([^"']+)["'][^>]*>/) do |match|
  full_tag = match
  src = $1
  
  # Remove srcset
  full_tag = full_tag.gsub(/srcset=["'][^"']+["']/, '')
  
  unless src.start_with?("data:")
    new_path = download_file(src, ASSET_DIRS["image"])
    full_tag.sub(src, new_path)
  else
    full_tag
  end
end

File.write(OUTPUT_FILE, content)
puts "Done! Saved to #{OUTPUT_FILE}"

File: scrape_assets.py
import os
import re
import urllib.request
from urllib.parse import urljoin, urlparse

# Configuration
BASE_URL = "https://www.anantawealth.com"
INPUT_FILE = "original_index.html"
OUTPUT_FILE = "index.html"
ASSET_DIRS = {
    "css": "css",
    "js": "js",
    "image": "images",
    "font": "fonts"
}

# Create directories
for dir_name in ASSET_DIRS.values():
    os.makedirs(dir_name, exist_ok=True)

def download_file(url, folder, default_ext=""):
    try:
        # Handle protocol-relative URLs
        if url.startswith("//"):
            url = "https:" + url
        
        # Absolute URL
        full_url = urljoin(BASE_URL, url)
        
        # Parse filename
        parsed = urlparse(full_url)
        path = parsed.path
        filename = os.path.basename(path)
        
        # Handle query parameters in filename (e.g. style.css?ver=1.0)
        # We want to keep unique versions to avoid conflicts, or just strip query?
        # Let's keep the filename simple but unique enough.
        
        if not filename or filename == "":
            return url 
            
        # Basic cleaning of filename
        filename = re.sub(r'[<>:"/\\|?*]', '_', filename)
        
        if not os.path.splitext(filename)[1] and default_ext:
            filename += default_ext

        local_path = os.path.join(folder, filename)
        
        if os.path.exists(local_path):
            # print(f"Skipping existing: {filename}")
            return f"{folder}/{filename}"

        print(f"Downloading: {full_url} -> {local_path}")
        
        req = urllib.request.Request(
            full_url, 
            headers={'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7)'}
        )
        with urllib.request.urlopen(req, timeout=10) as response, open(local_path, 'wb') as out_file:
            out_file.write(response.read())
            
        return f"{folder}/{filename}"
        
    except Exception as e:
        print(f"Failed to download {url}: {e}")
        return url 

def process_html():
    with open(INPUT_FILE, "r", encoding="utf-8") as f:
        content = f.read()

    # Regex patterns for finding assets
    # These are simplified and might miss some edge cases but should cover most WP sites.
    
    # 1. CSS Links: <link ... href="...">
    def replace_css(match):
        full_tag = match.group(0)
        href_match = re.search(r'href=["\']([^"\']+)["\']', full_tag)
        if href_match:
            original_url = href_match.group(1)
            # Filter for CSS or image-like things typical in link tags
            if ".css" in original_url or "rel='stylesheet'" in full_tag or 'rel="stylesheet"' in full_tag:
                 new_path = download_file(original_url, ASSET_DIRS["css"], ".css")
                 return full_tag.replace(original_url, new_path)
        return full_tag

    content = re.sub(r'<link[^>]+>', replace_css, content)

    # 2. Scripts: <script ... src="...">
    def replace_script(match):
        full_tag = match.group(0)
        src_match = re.search(r'src=["\']([^"\']+)["\']', full_tag)
        if src_match:
            original_url = src_match.group(1)
            new_path = download_file(original_url, ASSET_DIRS["js"], ".js")
            return full_tag.replace(original_url, new_path)
        return full_tag

    content = re.sub(r'<script[^>]+>', replace_script, content)
    
    # 3. Images: <img ... src="...">
    def replace_img(match):
        full_tag = match.group(0)
        
        # Remove srcset if present to force browser to use src (simpler for replication)
        full_tag = re.sub(r'srcset=["\'][^"\']+["\']', '', full_tag)
        
        src_match = re.search(r'src=["\']([^"\']+)["\']', full_tag)
        if src_match:
            original_url = src_match.group(1)
            if not original_url.startswith("data:"): # Skip data URIs
                new_path = download_file(original_url, ASSET_DIRS["image"])
                return full_tag.replace(original_url, new_path)
        return full_tag
        
    content = re.sub(r'<img[^>]+>', replace_img, content)

    # Write output
    with open(OUTPUT_FILE, "w", encoding="utf-8") as f:
        f.write(content)
    
    print(f"Done! Saved to {OUTPUT_FILE}")

if __name__ == "__main__":
    process_html()
